GitUI/UserControls/RevisionGrid.cs

```diff
@@ -2383,20 +2383,24 @@ private void CherryPickCommitToolStripMenuItemClick(object sender, EventArgs e)
 
         private void FixupCommitToolStripMenuItemClick(object sender, EventArgs e)
         {
-            if (Revisions.RowCount <= LastRowIndex || LastRowIndex < 0)
+            if (ShouldStartDialog())
                 return;
 
             UICommands.StartFixupCommitDialog(this, GetRevision(LastRowIndex));
         }
 
         private void SquashCommitToolStripMenuItemClick(object sender, EventArgs e)
         {
-            if (Revisions.RowCount <= LastRowIndex || LastRowIndex < 0)
+            if (ShouldStartDialog())
                 return;
 
             UICommands.StartSquashCommitDialog(this, GetRevision(LastRowIndex));
         }
 
+        private bool ShouldStartDialog()
+        {
+            return Revisions.RowCount <= LastRowIndex || LastRowIndex < 0;
+        }
```

Comment:
In some cases we cannot edit a commit with fixup...
For example, you have 3 commits : A, B, C.
You want fix the commit A with some change in the commit B.

It's just a case there is some other when editing is useful.

About renaming a commit with git extension : 
-> Click right on the commit
-> Click on "rebase on current branch...."
-> Click on "show options"
-> Check "Interactive rebase"
-> Click on rebase
-> edit the rebase interactive file : change pick by r
-> Save the file
-> close it
-> Here we go we can rename...

With Rename functionality in Advanced menu:
-> Click right on the commit
-> click on "edit commit"
-> Here we go we can rename

There is simply too much click to rename a commit...

In my company we care about commit message like a lot
and with gerrit review, we have a lot reword to do.
I converted a lot of people to git extension but they start complaining about that.
Moreover there is a plugin in eclipse which does this functionality...